sensor/debugger/k8s/fake.go

```diff
@@ -216,6 +219,188 @@ func (f *FakeEventsManager) Init() {
 		cronJobKind:               &batchv1.CronJob{},
 		podKind:                   &corev1.Pod{},
 	}
+	f.resourceListMap = map[string]func(reflect.Value, func(string) error) error{
```

Comment:
If I understand this code correctly, the only reason why we have this map is to call `GetName()` on each element on the `Items` slice. Since every resource list has the same interface (`.Items` + `.GetName()`), and we're only interested in the `string` value from the resource itself, I suggest that we make this generic. 

[Here's an example from playground](https://go.dev/play/p/90kOgXPFkWU). 

I suggest writing a single reflect-based function that iterates on `.Items` and calls `.GetName()` on each element:
```golang
func getNameGeneric(obj reflect.Value, f func(string) error) error {
	errorList := errorhelpers.NewErrorList("resource list function")
	items := obj.FieldByName("Items")
	if !items.IsValid() {
		errorList.AddError(errors.New("invalid kind: missing `Items` property"))
	} else {
		for i := 0; i < items.Len(); i++ {
			getNameFn := items.Index(i).MethodByName("GetName")
			returnVals := getNameFn.Call([]reflect.Value{})
			if err := f(returnVals[0].String()); err != nil {
				errorList.AddError(err)
			}
		}	
	}
	return errorList.ToError()
}
```

Reply:
Yeah. I changed your suggestion a little bit. Let me know what you think.